lib/rouge/lexers/powershell.rb

```diff
@@ -652,28 +97,47 @@ class Powershell < Shell
         rule %r/\s+/, Text
         rule %r/\$?"/, Str::Double, :double_quotes
         rule %r/\$'/, Str::Single, :ansi_string
-
         rule %r/'/, Str::Single, :single_quotes
-
         rule %r/\*/, Keyword
-
         rule %r/;/, Text
         rule %r/[^=\*\s{}()$"'`<]+/, Text
         rule %r/\d+(?= |\Z)/, Num
         rule %r/</, Text
         mixin :interp
       end
 
+      state :hashtable do 
+        rule %r/\w/, Name::Variable
+        rule %r/}/, Operator, :pop!
+        mixin :root
+      end
+
+      state :multiline do
+        rule %r/\.(#{MULTILINEKEYWORDS})/i, Keyword::Pseudo
+        rule %r/[\w,\d,\s,\.,\-,\,:\/,{,},<,>"*]/, Comment
+        rule %r/#>/, Comment, :pop!
+        mixin :root
+      end
+
+      state :heredoc do
+        rule %r/"@/, Operator, :pop!
+        rule %r/[^$\n]+/, Str::Heredoc
+        rule %r/[$]/, Str::Heredoc
+        mixin :data
+      end
+
       prepend :basic do
-        rule %r(<#[\s,\S]*?#>)m, Comment::Multiline
-        rule %r/#.*$/, Comment::Single
+        rule %r(#requires\s-version \d.\d*$),Comment::Preproc
```

Comment:
Can this appear anywhere in the file or only at the very top?

Reply:
> Can this appear anywhere in the file or only at the very top?

```#Requires``` can appear anywhere in the file. Per [PowerShell docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_requires?view=powershell-6):

> The ```#Requires``` statements can appear on any line in a script.